src/transformers/models/gpt_bigcode/modeling_gpt_bigcode.py

```diff
@@ -363,13 +363,6 @@ def forward(
 
         attn_dropout = self.attn_pdrop if self.training else 0.0
 
-        softmax_dtype = torch.float32 if self.attention_softmax_in_fp32 else query.dtype
```

Comment:
I think this block should be kept to ensure numerical correctness between the eager and fa2 version no? I don't see how (1 /( layer_idx + 1 )) * head_dim ** -0.5  can be identical  to (head_dim ** -0.5) (default of FA2)

Reply:
In eager, you then unscale by (layer_id+1), so the scaling used in eager is same as 1/sqrt(head_dim)

Reply:
Having this was the whole reason for FA2 giving gibberish generations.

Reply:
indeed - for reference: https://github.com/huggingface/transformers/blob/1c31b7aa3bb4e7ef24c77596d2a76f45a770159f/src/transformers/models/gpt_bigcode/modeling_gpt_bigcode.py#L199C12-L199C18
that's very strange why I did not get this when testing it out